src/Raven.Studio/typescript/models/database/tasks/replicationAccessBaseModel.ts

```diff
@@ -0,0 +1,112 @@
+﻿/// <reference path="../../../../typings/tsd.d.ts"/>
+import replicationCertificateModel = require("models/database/tasks/replicationCertificateModel");
+import prefixPathModel = require("models/database/tasks/prefixPathModel");
+import jsonUtil = require("common/jsonUtil");
+
+class replicationAccessBaseModel {
+    replicationAccessName = ko.observable<string>();
+    
+    certificate = ko.observable<replicationCertificateModel>();
+    
+    inputPrefixHubToSink = ko.observable<prefixPathModel>(new prefixPathModel(null));
+    inputPrefixSinkToHub = ko.observable<prefixPathModel>(new prefixPathModel(null));
+    
+    hubToSinkPrefixes = ko.observableArray<prefixPathModel>([]);
+    sinkToHubPrefixes = ko.observableArray<prefixPathModel>([]);
+    
+    samePrefixesForBothDirections = ko.observable<boolean>(false);
+    filteringPathsRequired = ko.observable<boolean>(true);
+        
+    dirtyFlag = new ko.DirtyFlag([]);
+
+    constructor(accessName: string, certificate: replicationCertificateModel, hubToSink: prefixPathModel[], sinkToHub: prefixPathModel[], filteringPathsRequired: boolean = true) {
+       
+        this.replicationAccessName(accessName);
+        this.hubToSinkPrefixes(hubToSink);
+        this.sinkToHubPrefixes(sinkToHub);
+        this.certificate(certificate);
+      
+        this.samePrefixesForBothDirections(_.isEqual(hubToSink.map(x => x.path()), sinkToHub.map(x => x.path())));
+       
+        this.filteringPathsRequired(filteringPathsRequired);
+    }
+    
+    initObservables() {
+        this.dirtyFlag = new ko.DirtyFlag([
+            this.replicationAccessName,
+            this.certificate,
+            this.hubToSinkPrefixes,
+            this.sinkToHubPrefixes,
+            this.samePrefixesForBothDirections
+        ], false, jsonUtil.newLineNormalizingHashFunction);
+    }
+    
+    initValidation() {
+        this.replicationAccessName.extend({
+           required: true
+        });
+        
+        this.certificate.extend({
+            required: true
+        })
+        
+        this.hubToSinkPrefixes.extend({
+            validation: [
+                {
+                    validator: () => !this.filteringPathsRequired() || this.hubToSinkPrefixes().length,
+                    message: "Please add at least one filtering prefix path."
+                }
+            ]
+        })
+
+        this.sinkToHubPrefixes.extend({
+            validation: [
+                {
+                    validator: () => !this.filteringPathsRequired() || this.samePrefixesForBothDirections() || this.sinkToHubPrefixes().length,
+                    message: "Please add at least one filtering prefix path, or use the Hub to Sink prefixes."
+                }
+            ]
+        })
+    }
+
+    addHubToSinkInputPrefixWithBlink() {
+        if (!this.hubToSinkPrefixes().filter(prefix => prefix.path() === this.inputPrefixHubToSink().path()).length) {
```

Comment:
can we use find instead of filter? No need to scan all - we should break on first not matching item.

Reply:
done